jsettlers.logic/src/jsettlers/logic/buildings/stack/multi/MultiRequestStack.java

```diff
@@ -0,0 +1,237 @@
+	 *            The position the stack will be.
+	 * @param buildingType
+	 */
+	public MultiRequestStack(IRequestsStackGrid grid, ShortPoint2D position, EBuildingType buildingType, EPriority priority,
+			MultiRequestStackSharedData sharedData) {
+		this.grid = grid;
+		this.position = position;
+		this.buildingType = buildingType;
+
+		this.sharedData = sharedData;
+
+		for (int i = 0; i < EMaterialType.NUMBER_OF_DROPPABLE_MATERIALS; i++) {
+			EMaterialType materialType = EMaterialType.DROPPABLE_MATERIALS[i];
+			materialRequests[materialType.ordinal] = new RequestStackOfMultiRequestStack(materialType);
+			grid.request(materialType, materialRequests[materialType.ordinal]);
+		}
+	}
+
+	void checkIfCurrentMaterialShouldBeReset() {
+		if (currentMaterialType != null && materialRequests[currentMaterialType.ordinal].getInDelivery() <= 0 && getStackSize() == 0) {
+			sharedData.unregisterHandlingStack(currentMaterialType, this);
+			currentMaterialType = null;
+		}
+	}
+
+	@Override
+	public boolean hasMaterial() {
+		return currentMaterialType != null && grid.hasMaterial(position, currentMaterialType);
+	}
+
+	/**
+	 * Pops a material from this stack. The material is of the type returned by {@link #getMaterialType()} and specified in the constructor.
+	 * 
+	 * @return <code>true</code> if there was a material to be popped from this stack. False otherwise.
+	 */
+	@Override
+	public boolean pop() {
+		if (currentMaterialType != null && grid.popMaterial(position, currentMaterialType)) {
+			popped++;
+			checkIfCurrentMaterialShouldBeReset();
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * This method gives the number of popped materials.
+	 * <p />
+	 * Due to the size of the variable, this method should only be used on limited stacks. Unlimited stacks my run into an overflow of the popped
+	 * value.
+	 * 
+	 * @return Returns the number of materials popped from this stack.
+	 */
+	@Override
+	public short getNumberOfPopped() {
+		return popped;
+	}
+
+	/**
+	 * Checks if all needed materials have been delivered. Therefore this method is only useful with bounded request stacks.
+	 * 
+	 * @return Returns true if this is a bounded stack and all the requested material has been delivered, <br>
+	 *         false otherwise.
+	 */
+	@Override
+	public boolean isFullfilled() {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public EMaterialType getMaterialType() {
+		return currentMaterialType;
+	}
+
+	@Override
+	public void releaseRequests() {
+		for (EMaterialType materialType : EMaterialType.DROPPABLE_MATERIALS) {
+			sharedData.requestedMaterials[materialType.ordinal] = 0;
+			grid.createOffersForAvailableMaterials(position, materialType);
+		}
+		released = true;
+	}
+
+	@Override
+	public int getStackSize() {
+		return currentMaterialType == null ? 0 : grid.getStackSize(position, currentMaterialType);
+	}
+
+	@Override
+	public short getStillRequired() {
+		throw new UnsupportedOperationException();
```

Comment:
The problem is that this type of stack cannot support this method. In the case that no currentMaterialType is set, it is unclear what to return here. However, as this type of stack is never used during construction (which is the user of this method), it is not a problem that this method is not supported here.

Therefore I thought UnsupportedOperationException is more correct than "not implemented yet".